File: packages/urdu-js/tests/normalization/kashida.test.ts
import { describe, it, expect } from 'vitest'
import { removeKashida } from '../../src/normalization/kashida.js'

describe('removeKashida', () => {
  it('empty', () => expect(removeKashida('')).toBe(''))
  it('removes tatweel', () => expect(removeKashida('ک\u0640تاب')).toBe('کتاب'))
  it('no kashida unchanged', () => expect(removeKashida('کتاب')).toBe('کتاب'))
  it('only kashida → empty', () => expect(removeKashida('\u0640\u0640\u0640')).toBe(''))
  it('multiple kashidas', () => expect(removeKashida('ک\u0640\u0640تاب')).toBe('کتاب'))
})

File: packages/urdu-dart/test/normalization/kashida_test.dart
import 'package:test/test.dart';
import 'package:urdu_tools/urdu_tools.dart';

void main() {
  group('removeKashida', () {
    test('empty', () => expect(removeKashida(''), equals('')));
    test('removes tatweel',
        () => expect(removeKashida('کـتاب'), equals('کتاب')));
    test('no kashida unchanged',
        () => expect(removeKashida('کتاب'), equals('کتاب')));
    test('only kashida -> empty',
        () => expect(removeKashida('\u0640\u0640\u0640'), equals('')));
    test('multiple kashidas',
        () => expect(removeKashida('کــتاب'), equals('کتاب')));
  });
}
